src/SubPoolFactory.sol

```diff
@@ -0,0 +1,193 @@
+    error SubPoolFactory__SolverNotAMember();
+
+    event UpdateExitDelay(uint256 newDelay);
+    event UpdateBackendUri(address indexed pool, string uri);
+    event SolverPoolDeployed(address indexed solver, address indexed pool);
+    event SolverPoolBilled(address indexed pool, uint256 amt, uint256 cowAmt, uint256 ethAmt, string reason);
+    event AnnounceExit(address indexed pool);
+    event UpdateSolverMembership(address indexed pool, address indexed solver, bool isMember);
+
+    struct SubPoolData {
+        /// The collateral token that pool was initialized with.
+        address collateral;
+        /// The timestamp at which the pool can exit.
+        uint176 exitTimestamp;
+    }
+
+    /// @notice Individual subpool data
+    mapping(address => SubPoolData) public subPoolData;
+    /// @notice Solver backend uri
+    /// @dev    Not stored in the SubPoolData because we don't want to load the string
+    ///         everytime we read the subpooldata in memory.
+    mapping(address => string) public backendUri;
+    /// @notice One to one mappping of solver to a subpool.
+    /// @dev    If a solver already has a subpool of their own, they cannot belong to any other pool.
+    mapping(address => address) public solverBelongsTo;
+    /// @notice the delay between `announceExit` and `exit`.
+    uint256 public exitDelay;
+    address public immutable COW;
+
+    constructor(uint256 exitDelay_, address cow) {
+        exitDelay = exitDelay_;
+        emit UpdateExitDelay(exitDelay);
+        COW = cow;
+        _addOwner(msg.sender);
+    }
+
+    /// @notice Set the delay between `announceExit` and being able to withdraw funds.
+    function setExitDelay(uint32 delay) external auth {
+        exitDelay = delay;
+        emit UpdateExitDelay(delay);
+    }
+
+    /// @notice Create a `SubPool` for the user at a deterministic address with salt as `msg.sender`.
+    /// @param token  - The token to use as collateral.
+    function create(address token, uint256 amt, uint256 cowAmt, string calldata uri) external returns (address) {
+        SubPool subpool = new SubPool{salt: bytes32(0)}(msg.sender, COW);
+        // initialization needs to happen after deployment to prevent collateralToken being part of the initcode
+        // to prevent multiple pools per solver, initcode needs to be static for a given solver
+        subpool.initializeCollateralToken(token);
+        emit SolverPoolDeployed(msg.sender, address(subpool));
+
+        subPoolData[address(subpool)] = SubPoolData({collateral: token, exitTimestamp: 0});
+        backendUri[address(subpool)] = uri;
+        emit UpdateBackendUri(address(subpool), uri);
+
+        // the pool creator becomes a member of its pool by default so it doesn't require another
+        // manual transaction to set oneself as solver
+        solverBelongsTo[msg.sender] = address(subpool);
+        emit UpdateSolverMembership(address(subpool), msg.sender, true);
```

Comment:
There's no `UpdateSolverMembership(..., false)` in case the creator currently belongs to another pool.
In general, I don't think we should add the creator to the solvers by default as I expect most projects wouldn't want to have the private key they use for sending out settlements in a script be the same that can update the subpool owners and take the funds. (Nor we want to imply that this the "expected thing to do".)

Reply:
removed the default